docs/dev/framework/security.md

```diff
@@ -0,0 +1,275 @@
+(as defined via the DCA), folders and modules for example:
+
+```php
+// User has access to form ID 5
+$security->isGranted('contao_user.forms', 5);
+
+// User is allowed to access field "published" of table "tl_page"
+$security->isGranted('contao_user.alexf', 'tl_page::published');
+
+// Check access to folder
+$security->isGranted('contao_user.filemounts', '/files/foo/bar');
+```
+
+{{< version "4.10" >}}
+
+You can also use the security helper to check wether the user can edit a page or is allowed to edit fields in a specific DCA for example:
+
+```php
+// whether the user can access any field in tl_page
+$security->isGranted('contao_user.can_edit_fields', 'tl_page');
+
+// whether user can edit the given page (array of row or page model)
+$security->isGranted('contao_user.can_edit_page', [/* row of page data */]);
+$security->isGranted('contao_user.can_edit_page', $pageModel);
+```
+
+{{< version "4.12" >}}
+
+The security helper can also be used to check whether a front end or back end user belongs to any of the specified user groups:
+
+```php
+$security->isGranted('contao_group', $groupId);
+$security->isGranted('contao_group', [/* array of group IDs */]);
+```
+
+{{% notice tip %}}
+Since Contao **4.10** there are class constants available for the various permission attributes, so that you do not have to remember them
+yourself and instead can use your IDE to find the correct attribute. For the Contao Core these constants are available in 
+`Contao\CoreBundle\Security\ContaoCorePermissions` while the permissions of the additional bundles are available in 
+`Contao\NewsBundle\Security\ContaoNewsPermissions`, `Contao\CalendarBundle\Security\ContaoCalendarPermissions`, 
+`Contao\NewsletterBundle\Security\ContaoNewsletterPermissions` and `Contao\FaqBundle\Security\ContaoFaqPermissions` respectively. The
+PHP documentation of these constants also contain hints about what type the subject should be. Examples:
+
+```php
+use Contao\CoreBundle\Security\ContaoCorePermissions;
+use Contao\NewsBundle\Security\ContaoNewsPermissions;
+
+// Whether user can access the news module in the back end
+$security->isGranted(ContaoCorePermissions::USER_CAN_ACCESS_MODULE, 'news');
+
+// Whether user can use the hidden input field in the form generator
+$security->isGranted(ContaoCorePermissions::USER_CAN_ACCESS_FIELD, 'hidden');
+
+// Whether user has access to any field of tl_content
+$security->isGranted(ContaoCorePermissions::USER_CAN_EDIT_FIELDS_OF_TABLE, 'tl_content');
+
+// Whether user can create news archives
+$security->isGranted(ContaoNewsPermissions::USER_CAN_CREATE_ARCHIVES);
+```
+{{% /notice %}}
+
+
+## Custom Back End Access Rights
+
+To implement your own back end access rights (e.g. for custom modules in the back end) the following steps are necessary:
+
+1. Add your new permission to `$GLOBALS['TL_PERMISSIONS']`.
```

Comment:
maybe we need to explain what this does?

Reply:
Yeah, I would add that adding your permission field to `TL_PERMISSIONS` enables Contao's `contao_user` voter to check for this permission.